src/Exporter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry;
+
+use OpenTelemetry\Tracing\Span;
+use OpenTelemetry\Tracing\Tracer;
+
+abstract class Exporter
+{
+    abstract public function convertSpan(Span $span) : array;
```

Comment:
I would prefer that we don't have to convert to an array of arrays, and then almost certainly have to do *another* conversion to transport it. Is this approach mandated/suggested by the specification? I don't see it in the Tracing API -- is it somewhere else maybe?

Reply:
After thinking about it more, I realized an implementation could override the `flush` function as well, so maybe it's a non-issue. This way it could detect concrete `Transports` and see if they can handle direct conversions from `Span`s in a vendor specific API.

Reply:
The [Tracing SDK Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/c9d3d439492e07ca900cf28f98552a8aff919ff5/specification/sdk-tracing.md) says this about the [Span Exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/c9d3d439492e07ca900cf28f98552a8aff919ff5/specification/sdk-tracing.md#span-exporter):

> The exporter must support two functions: Export and Shutdown. In strongly typed languages typically there will be 2 separate Exporter interfaces, one that accepts spans (SpanExporter) and one that accepts metrics (MetricsExporter).

This doesn't seem to follow the specification, unless there is yet another place I should be looking. I wouldn't discount that possibility, given how new I am to the project.

Reply:
I don't think there is another place.  I think we may want to follow the spec a little more deliberately here, which we could update in subsequent PRs.